File: LeetCode/src/MyLinkedList/Q21_合并两个有序链表.java
package MyLinkedList;

/**
 * @author Hyperspace
 * @date 2022/02/19
 * <p>
 * 1. 归并
 */
public class Q21_合并两个有序链表 {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode(0);
        ListNode cur = dummy;
        while (list1 != null && list2 != null) {
            if (list1.val <= list2.val) {
                cur.next = list1;
                list1 = list1.next;
            } else {
                cur.next = list2;
                list2 = list2.next;
            }
            cur = cur.next;
        }
        cur.next = list1 == null ? list2 : list1;

        return dummy.next;
    }
}

File: GolangSolutions/LinkedList/Q21_合并两个有序链表.go
/*
	@author Hyper
	@date 2023/08/05
	@file Q21_合并两个有序链表.go
*/

package LinkedList

func mergeTwoLists(list1 *ListNode, list2 *ListNode) *ListNode {
	l := &ListNode{}
	cur := l
	for list1 != nil && list2 != nil {
		if list1.Val <= list2.Val {
			cur.Next = list1
			list1 = list1.Next
		} else {
			cur.Next = list2
			list2 = list2.Next
		}
		cur = cur.Next
	}
	if list1 != nil {
		cur.Next = list1
	}
	if list2 != nil {
		cur.Next = list2
	}

	return l.Next
}
